pkg/template/template.go

```diff
@@ -25,32 +27,71 @@ type (
 	// Repository defines all the helper functions to access the generated template specific types and fields.
 	Repository interface {
 		GetTemplateInfo(template string) (Info, bool)
+		DoesBuilderSupportsTemplate(hndlrBuilder adptConfig.HandlerBuilder, tmpl string) (bool, string)
 	}
 	// TypeEvalFn evaluates an expression and returns the ValueType for the expression.
 	TypeEvalFn func(string) (pb.ValueType, error)
 	// InferTypeFn does Type inference from the Constructor.params proto message.
 	InferTypeFn func(proto.Message, TypeEvalFn) (proto.Message, error)
 	// ConfigureTypeFn dispatches the inferred types to handlers
 	ConfigureTypeFn func(types map[string]proto.Message, builder *adptConfig.HandlerBuilder) error
+	// SupportsBuilderFn check if the handlerBuilder supports template.
+	SupportsBuilderFn func(hndlrBuilder adptConfig.HandlerBuilder) bool
 	// Info contains all the information related a template like
 	// Default constructor params, type inference method etc.
 	Info struct {
-		CnstrDefConfig  proto.Message
-		InferTypeFn     InferTypeFn
-		ConfigureTypeFn ConfigureTypeFn
+		CnstrDefConfig    proto.Message
+		InferTypeFn       InferTypeFn
+		ConfigureTypeFn   ConfigureTypeFn
+		SupportsBuilderFn SupportsBuilderFn
+		BuilderName       string
```

Comment:
It is the Builder name, name is confusing because template also has a name.
The builder name is used for error reporting.